src/configure/mod.rs

```diff
@@ -158,311 +103,312 @@ impl Context {
+    info!(ctx, "key generation completed");
+
+    progress!(ctx, 940);
+
+    Ok(())
+}
+
+#[derive(Debug, PartialEq, Eq)]
+enum AutoconfigProvider {
+    Mozilla,
+    Outlook,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+struct AutconfigSource {
+    provider: AutoconfigProvider,
+    url: String,
+}
+
+impl AutconfigSource {
+    async fn fetch(&self, ctx: &Context, param: &LoginParam) -> Result<LoginParam> {
+        let params = match self.provider {
+            AutoconfigProvider::Mozilla => moz_autoconfigure(ctx, &self.url, &param).await?,
+            AutoconfigProvider::Outlook => outlk_autodiscover(ctx, &self.url, &param).await?,
+        };
+
+        Ok(params)
+    }
 }
 
-#[derive(Debug)]
-enum Step {
-    Done,
-    Continue,
+/// Retrieve available autoconfigurations.
+///
+/// A Search configurations from the domain used in the email-address, prefer encrypted
+/// B. If we have no configuration yet, search configuration in Thunderbird's centeral database
+async fn get_autoconfig(
+    ctx: &Context,
+    param: &LoginParam,
+    param_domain: &str,
+    param_addr_urlencoded: &str,
+) -> Option<LoginParam> {
+    let mut param_autoconfig = None;
+
+    let urls = [
+        AutconfigSource {
+            provider: AutoconfigProvider::Mozilla,
+            url: format!(
+                "https://autoconfig.{}/mail/config-v1.1.xml?emailaddress={}",
+                param_domain, param_addr_urlencoded
+            ),
+        },
+        // the doc does not mention `emailaddress=`, however, Thunderbird adds it, see https://releases.mozilla.org/pub/thunderbird/ ,  which makes some sense
+        AutconfigSource {
+            provider: AutoconfigProvider::Mozilla,
+            url: format!(
+                "https://{}/.well-known/autoconfig/mail/config-v1.1.xml?emailaddress={}",
+                param_domain, param_addr_urlencoded
+            ),
+        },
+        AutconfigSource {
+            provider: AutoconfigProvider::Outlook,
+            url: format!("https://{}/autodiscover/autodiscover.xml", param_domain),
+        },
+        // Outlook uses always SSL but different domains (this comment describes the next two steps)
+        AutconfigSource {
+            provider: AutoconfigProvider::Outlook,
+            url: format!(
+                "https://{}{}/autodiscover/autodiscover.xml",
+                "autodiscover.", param_domain
+            ),
+        },
+        AutconfigSource {
+            provider: AutoconfigProvider::Mozilla,
+            url: format!(
+                "http://autoconfig.{}/mail/config-v1.1.xml?emailaddress={}",
+                param_domain, param_addr_urlencoded
+            ),
+        }, // do not transfer the email-address unencrypted
+        AutconfigSource {
+            provider: AutoconfigProvider::Mozilla,
+            url: format!(
+                "http://{}/.well-known/autoconfig/mail/config-v1.1.xml",
+                param_domain
+            ),
+        },
```

Comment:
I wonder why we still do http-requests -- it's unsafe and probably shouldn't be done.(autoconfigs can point to arbtirary servers and we are sending login-credentials there).

Reply:
I would be more than happy to remove them, but didn't want to make that call by myself
cc @r10s

Reply:
well, https was added 2019 to the mozilla-spec and mozilla wants to keep supporting http up to 2023, see https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird/Autoconfiguration#SSL

but i would not insist on following that way, maybe just remove it and see what will happen.

Reply:
removed